pkg/install/install.go

```diff
@@ -128,46 +160,46 @@ func (i *Installer) Install(ctx context.Context, installConfig *installconfig.In
 	}
 
 	for {
-		i.log.Printf("starting phase %s", i.doc.OpenShiftCluster.Properties.Install.Phase)
 		switch i.doc.OpenShiftCluster.Properties.Install.Phase {
 		case api.InstallPhaseDeployStorage:
-			err := i.installStorage(ctx, installConfig, platformCreds, image)
-			if err != nil {
-				return err
-			}
-
 		case api.InstallPhaseDeployResources:
-			err := i.installResources(ctx)
-			if err != nil {
-				return err
-			}
-
 		case api.InstallPhaseRemoveBootstrap:
-			err := i.removeBootstrap(ctx)
-			if err != nil {
-				return err
-			}
-
-			i.doc, err = i.db.PatchWithLease(ctx, i.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
-				doc.OpenShiftCluster.Properties.Install = nil
-				return nil
-			})
-			return err
-
 		default:
 			return fmt.Errorf("unrecognised phase %s", i.doc.OpenShiftCluster.Properties.Install.Phase)
 		}
-
-		i.doc, err = i.db.PatchWithLease(ctx, i.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
-			doc.OpenShiftCluster.Properties.Install.Phase++
-			return nil
-		})
-		if err != nil {
-			return err
+		i.log.Printf("starting phase %s", i.doc.OpenShiftCluster.Properties.Install.Phase)
+		for _, step := range steps[i.doc.OpenShiftCluster.Properties.Install.Phase] {
+			i.log.Print(step.progress)
+			err := step.run(ctx)
+			if i.doc.OpenShiftCluster.Properties.Install == nil && err == nil {
+				// sucessful end of the install process
+				return nil
+			}
```

Comment:
warning - you're in the same area as @mjudeikis with #154 - please sync

Reply:
I think we will have a code conflict, but not because of a logical conflict, just because he is removing the loop.

Reply:
correct.